napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,521 @@
+                axis=0,
+            )
+            self.color_properties = ColorProperties(
+                name=color_property_name,
+                values=new_color_property_values,
+                current_value=current_value,
+            )
+
+            if update_clims and self.mode == ColorMode.COLORMAP:
+                self.contrast_limits = None
+
+    def remove(self, indices_to_remove: Union[set, list, np.ndarray]):
+        """Remove the indicated color elements
+        Parameters
+        ----------
+        indices_to_remove : set, list, np.ndarray
+            The indices of the text elements to remove.
+        """
+        selected_indices = list(indices_to_remove)
+        if len(selected_indices) > 0:
+            if self.mode == ColorMode.DIRECT:
+                self.colors = np.delete(self.colors, selected_indices, axis=0)
+            else:
+                # remove the color_properties
+                color_property_name = self.color_properties.name
+                current_value = self.color_properties.current_value
+                new_color_property_values = np.delete(
+                    self.color_properties.values, selected_indices
+                )
+                self.color_properties = ColorProperties(
+                    name=color_property_name,
+                    values=new_color_property_values,
+                    current_value=current_value,
+                )
+
+    def paste(self, colors: np.ndarray, properties: Dict[str, np.ndarray]):
+        """Append colors to the ColorManager. Uses the color values if
+        in direct mode and the properties in colormap or cycle mode.
+
+        This method is for compatibility with the paste functionality
+        in the layers.
+
+        Parameters
+        ----------
+        colors : np.ndarray
+            The (Nx4) color array of color values to add. These values are
+            only used if the color mode is direct.
+        properties : Dict[str, np.ndarray]
+            The property values to add. These are used if the color mode
+            is colormap or cycle.
+        """
+        if self.mode == ColorMode.DIRECT:
+            self.colors = np.concatenate(
+                (self.colors, transform_color(colors))
+            )
+        else:
+            color_property_name = self.color_properties.name
+            current_value = self.color_properties.current_value
+            old_properties = self.color_properties.values
+            values_to_add = properties[color_property_name]
+            new_color_property_values = np.concatenate(
+                (old_properties, values_to_add),
+                axis=0,
+            )
+
+            self.color_properties = ColorProperties(
+                name=color_property_name,
+                values=new_color_property_values,
+                current_value=current_value,
+            )
+
+    def update_current_properties(
```

Comment:
Maybe make private if generally called by layer and we don't expect users to call it?

Reply:
Done!